js/Utility.html

```diff
@@ -12,6 +12,77 @@
   array[index] = array[index].replace(/\s/g, '');
 },
 
+utility.stripItem = (value, index, array) => {
+  // only the quantity and barcode are needed, because the integrity
+  // check will pull the other information from the cache
+  array[index] = {
+    id: array[index].id,
+    description: array[index].description,
+    quantity: array[index].quantity,
+    barcode: array[index].barcode,
+  };
+},
+
+utility.itemIntegrityCheck = (item, index, array) => {
+  const items = app.cache.items;
+  const manualItemIdregex = /^MANUAL/;
+  var found = false;
```

Comment:
```suggestion
  let found = items.find(i => i.barcode == item.barcode);
```
This eliminates the following loop.

Reply:
yes, this is what i did!